### PR TITLE
Auto/update behavior velocity doc

### PR DESCRIPTION
**Pull request by taikitanaka3**
## Remarks

occlusion spot
includes this update https://github.com/tier4/autoware.iv/pull/2283

intersection
https://github.com/tier4/autoware.iv/pull/2274
https://github.com/tier4/autoware.iv/pull/2294
https://github.com/tier4/autoware.iv/pull/1958
https://github.com/tier4/autoware.iv/pull/1990

walkway
https://github.com/tier4/AutowareArchitectureProposal.iv/pull/746

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [pull request guidelines][pull-request-guidelines]
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by taikitanaka3** [comment]
@TakaHoribe 
can you review until this commit?

File: planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/merge-from-private-design.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Walkway
+
+### Role
+
+私有地（例えば駐車場）から公道へ出る際には面出しする必要がある. もし交差点のレーンに私有地タグ(location = private)がついていて、次のレーンが私有地ではない場合は必ず一時停止をして交差点に侵入する. 停止位置の算出方法は交差点モジュールと同じで、注視領域も同じ
```
Comment (TakaHoribe) [inline planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/merge-from-private-design.md:5]: @taikitanaka3 この役割の説明って合ってますか？「私有地から公道に出るとき」ではなく、「歩道を通過するとき」かと思ったのだけど。「私有地から公道に出るとき」はmerge_from_private_areaではない？

File: planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/occlusion-spot-design.md [hunk 2]
```diff
@@ -1,29 +1,71 @@
...
+This module is activated when the ego-lane has a private/public attribute.
 
-#### Occlusion Spot Public
+### Limitation
 
-This module only works in public road and use dynamic objects to detect occlusion spots.
+`point cloud` vs `occupancy grid` vs `object detection`
 
-Considering all occupancy grid cells inside focus range requires a lot of computation cost, so this module ignores to search farther occlusion spot which is longitudinally or laterally slice once occlusion spot is found.
+このモジュールでは`occupancy grid` と `object detection`の情報を使用している。安全を担保するために point cloud の生の情報を利用する方法もあるが、その点群が草なのか、落ち葉なのか、柱なのか、信号待ちの車両によるものなのかの判定がつかない。そこでこのモジュールでは Occupancy Grid を用いて死角と判定する際に死角の大きさ、死角と自分のレーンとの間に専有グリッドがないかどうかなどの判定を入れている。これによって死角の誤検知は減った。しかしながら植え込みや柵の形状などによっては OccupancyGrid の生成がうまく行かないシーンもありその場合は不要な減速が入ってしまうこともある。こちらは OccupancyGrid の改善が課題となっている。また、公道では信号待ちの車両や、渋滞しているレーンにいる車両など人が飛び出てくるとは考えづらいシーンで減速をかけないために、減速対象の死角を作成する車両が路駐しているのかどうかなど判定をいれている。
```
Comment (TakaHoribe) [inline planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/occlusion-spot-design.md:17]: @taikitanaka3 occupancy_gridってほぼ生の点群情報ですよね？

File: planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/merge-from-private-design.md (hunk 1 above)
Comment (taikitanaka3) [reply planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/merge-from-private-design.md:5]: タイトルがmerge_from_privateでしたすんません

File: planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/occlusion-spot-design.md [hunk 3]
```diff
@@ -1,29 +1,64 @@
...
+
+### Inner-workings / Algorithms
 
-##### Definition
+#### Logics Working In Private/Public Load
 
-This module insert safe velocity at collision point and show virtual wall at intersection below.
+Deceleration for the occlusion spot works on **different logic** depending on whether the driving road is **public or private**.
 
+There are several types of occlusions, such as "occlusions generated by parked vehicles" and "occlusions caused by obstructions". In situations such as driving on **private roads**, where people jump out of the way frequently, all possible occlusion spots must be taken into account. Therefore, on private roads, the deceleration plan considers all occlusion spots calculated from the **occupancy grid**. On the other hand, while driving at high speeds, it is not reasonable to take into account all occlusion spots (e.g., jumping out from behind a jumping out from behind a guardrail). Therefore, on **public roads**, the target of the deceleration calculation is currently limited to only the occlusion spots generated by vehicles parked on the road shoulder, which uses the **dynamic object** information.
```
Comment (taikitanaka3) [inline planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/occlusion-spot-design.md:29]:
```suggestion
There are several types of occlusions, such as "occlusions generated by parked vehicles" and "occlusions caused by obstructions". In situations such as driving on **private roads**, where people jump out of the way frequently, all possible occlusion spots must be taken into account. Therefore, on private roads, the deceleration plan considers all occlusion spots calculated from the **occupancy grid**. On the other hand, while driving at high speeds, it is not reasonable to take into account all occlusion spots (e.g., jumping out from behind a guardrail). Therefore, on **public roads**, the target of the deceleration calculation is currently limited to only the occlusion spots generated by vehicles parked on the road shoulder, which uses the **dynamic object** information.
```

File: planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/merge-from-private-design.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+## Walkway
```
Comment (taikitanaka3) [inline planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/merge-from-private-design.md:1]:
```suggestion
## Merge From Private
```

File: planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/merge-from-private-design.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+### Role
+
+私有地（例えば駐車場）から公道へ出る際には面出しする必要がある. もし交差点のレーンに私有地タグ(location = private)がついていて、次のレーンが私有地ではない場合は必ず一時停止をして交差点に侵入する. 停止位置の算出方法は交差点モジュールと同じで、注視領域も同じ
+i.e. 必ず一時停止を行う以外は交差点モジュールと同じ機能になる.
+
+![walkway](docs/intersection/merge_from_private.png)
```
Comment (taikitanaka3) [inline planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/merge-from-private-design.md:8]:
```suggestion
![merge-from-private](docs/intersection/merge_from_private.png)
```

**Comment by taikitanaka3** [comment]
@TakaHoribe 
ping

**Review by TakaHoribe — APPROVED** [review]

**Pull request merged**